providers/hexonet/hexonetProvider.go

```diff
@@ -0,0 +1,77 @@
+// Package hexonet implements a registrar that uses the hexonet api to set name servers. It will self register it's providers when imported.
+package hexonet
+
+// HEXONET is a leading developer and operator of domain names and DNS platforms.
+// Thousands of individuals, service providers and registrars globally choose HEXONET
+// for domains and DNS for our advanced technology, operational performance and up-time,
+// and most importantly for unparalleled expertise in DNS.
+// DnsControl with HEXONET's DNS marries DNS automation with an industry-leading DNS
+// platform that supports DNSSEC, PremiumDNS via Anycast Network, and nearly all of
+// DnsControl's listed provider features.
+
+import (
+	"encoding/json"
+
+	"github.com/StackExchange/dnscontrol/providers"
+	hxcl "github.com/hexonet/go-sdk/client"
+	"github.com/pkg/errors"
+)
+
+// HXClient describes a connection to the hexonet API.
+type HXClient struct {
+	APILogin    string
+	APIPassword string
+	APIEntity   string
+	client      *hxcl.Client
+}
+
+var features = providers.DocumentationNotes{
+	providers.CanUseAlias:            providers.Can("We fallback to CNAME up to now. Our API comes already with realtime ALIAS support over X-ALIAS-A / X-ALIAS-AAAA. This will be supported shortly."),
+	providers.CanUseCAA:              providers.Can(),
+	providers.CanUsePTR:              providers.Can(),
+	providers.CanUseRoute53Alias:     providers.Cannot(),
+	providers.CanUseSRV:              providers.Can(),
+	providers.CanUseTLSA:             providers.Can(),
+	providers.CanUseTXTMulti:         providers.Can(),
+	providers.CantUseNOPURGE:         providers.Can(),
+	providers.DocCreateDomains:       providers.Can(),
+	providers.DocDualHost:            providers.Can(),
+	providers.DocOfficiallySupported: providers.Can(),
+}
+
+func newProvider(conf map[string]string) (*HXClient, error) {
+	api := &HXClient{
+		client: hxcl.NewClient(),
+	}
+	api.APILogin, api.APIPassword, api.APIEntity = conf["apilogin"], conf["apipassword"], conf["apientity"]
+	if conf["debugmode"] == "1" {
+		api.client.EnableDebugMode()
+	}
+	if len(conf["ipaddress"]) > 0 {
+		api.client.SetIPAddress(conf["ipaddress"])
+	}
+	if api.APIEntity != "1234" && api.APIEntity != "54cd" {
+		return nil, errors.Errorf("wrong api system entity used. use \"1234\" for OT&E system or \"54cd\" for Live system")
```

Comment:
Prefer `fmt.Errorf` to the `errors` package directly. Mostly for consistency across the code base.

Reply:
makes sense